### PR TITLE
Bug/dl 48

### PR DESCRIPTION
**Pull request by pako535**

### CODE REVIEW

**Comment by oskam** [comment]
We shouldn't make pull requests into "release" branch?

**Comment by pako535** [comment]
@oskam yea my bad

File: views/index.py [hunk 1]
```diff
@@ -84,14 +86,23 @@ def registration():
...
+                    db.session.add(new_user)
+                    db.session.commit()
+                    send_confirmation_email(new_user.email)
+                else:
+                    message_body = 'User already exist'
+                    message_title = 'Opss!'
+                    return render_template('message.html',
+                                           message_title=message_title,
+                                           message_body=message_body)
             except:
```
Comment (etude-ist) [inline views/index.py]: As was previously said: this is bad practice. Let's add specific exeptions here.

File: views/index.py [hunk 2]
```diff
@@ -84,14 +86,23 @@ def registration():
...
+                        surname=form.surname.data,
+                        password_hash=generate_password_hash(
+                            form.password.data))
+                    db.session.add(new_user)
+                    db.session.commit()
+                    send_confirmation_email(new_user.email)
+                else:
+                    message_body = 'User already exist'
+                    message_title = 'Opss!'
+                    return render_template('message.html',
```
Comment (tdiak) [inline views/index.py]: I think we can move message_body and title to the static property in RegisterForm. Later we can use:
message_title=RegistrationForm.error_title etc.

File: views/index.py [hunk 3]
```diff
@@ -84,14 +86,23 @@ def registration():
...
-                new_user = User(
-                    email=form.email.data,
-                    first_name=form.first_name.data,
-                    surname=form.surname.data,
-                    password_hash=generate_password_hash(form.password.data))
-                db.session.add(new_user)
-                db.session.commit()
-                send_confirmation_email(new_user.email)
+                data = User.query.filter_by(email=form.email.data).first()
+                if data is None:
```
Comment (tdiak) [inline views/index.py]: Are You sure "is None" is obligatory ?
Maybe:
if User.query.filter_by(email=form.email.data).first():
....
will be ok.

We can create user_exists(email) static method in User class.
if not User.user_exists(...):
...

File: views/index.py (hunk 1 above)
Comment (tdiak) [reply views/index.py]: Yes, we should use specific exception

File: views/index.py [hunk 4]
```diff
@@ -84,15 +86,24 @@ def registration():
...
-                    first_name=form.first_name.data,
-                    surname=form.surname.data,
-                    password_hash=generate_password_hash(form.password.data))
-                db.session.add(new_user)
-                db.session.commit()
-                send_confirmation_email(new_user.email)
-            except:
+                if User.query.filter_by(email=form.email.data).first():
+                    message_body = 'User already exist'
+                    message_title = 'Opss!'
```
Comment (mariawedler) [inline views/index.py]: Oops

**Pull request merged**